compiler-core/src/erlang.rs

```diff
@@ -464,8 +464,32 @@ fn escape_atom_string(value: String) -> String {
     }
 }
 
+fn unicode_escape_sequence_pattern() -> &'static Regex {
+    static PATTERN: OnceLock<Regex> = OnceLock::new();
+    PATTERN.get_or_init(|| {
+        Regex::new(r#"\\(\\)?u"#).expect("Unicode escape sequence regex cannot be constructed")
+    })
+}
+
 fn string(value: &str) -> Document<'_> {
-    value.to_doc().surround("<<\"", "\"/utf8>>")
+    Document::String(
+        unicode_escape_sequence_pattern()
+            // `\\u`-s should not be affected, so that "\\u..." is not converted to
+            // "\\x...". That's why capturing groups is used to exclude cases that
+            // shouldn't be replaced.
+            .replace_all(value, |caps: &Captures<'_>| {
+                if caps.get(1).is_some() {
```

Comment:
I think this would not handle `\\\u{...}` correctly. We'd need to check how many slashes there are, rewriting if there's an odd number, not if there's an even number.